### ISSUE TITLE
how to use folder data VidCoder portable?

### ISSUE THREAD

**Issue by candrapersada** [RandomEngy/VidCoder#639]
how to use folder data VidCoder portable?

**Comment by RandomEngy** [comment]
Are you asking how to have it keep the settings in the same folder as the executable? It should do that automatically, unless you have one already in %appdata%\VidCoder\VidCoder.sqlite or %appdata%\VidCoder-Beta\VidCoder.sqlite

**Comment by amreus** [comment]
Hope you don't mind resurrecting an old issue.

>  It should do that automatically, unless you have one already in %appdata%\VidCoder\VidCoder.sqlite or %appdata%\VidCoder-Beta\VidCoder.sqlite

This isn't my experience so far.

I've been trying to get VidCoder to use/create a `VidCoder.sqlite` file in the executable folder but the file is always created in `%appdata%\VidCoder\VidCider.sqlite`.

I'm using `VidCoder-6.44-Portable.exe` and 7zip to extract the files, and running `vidcoder` from the command line. I'm just working out of my `Downloads` folder so file/folder permissions should be in order. It's not a terribly important issue to me but if you had any ideas. Thanks.

**Comment by RandomEngy** [comment]
VidCoder does a check to determine if it's running in Portable mode or not: If it's running in the Temp directory, it figures that it was put there by the self-extracting portable file and will activate the "portable" features. One of these features is using the current directory to store the VidCoder.sqlite database file.

Because you are manually extracting the files and presumably not extracting them to your system Temp folder, it doesn't have that behavior.

**Comment by amreus** [comment]
Thanks. So that means the Portable version deletes the config file along with everything else at exit.

**Comment by RandomEngy** [comment]
Anything in the temp folder would be deleted, but not the files in %appdata% like VidCoder.sqlite. The temp folder contains what would normally be in Program Files under the normal installer.

**Comment by BinaryRider** [comment]
Why did you choose this unusual way for the portable version?
I would also like to have VidCoder unpacked portable under C:\PortableApps\... . I also notice that you can't configure the file folder for the logs anymore either, as it was the case in the Version 5.

It would be really nice if this was all configurable and the program could recognize that it is portable by the presence of a file (portable.dat) in the program directory for example. The solution with the %TEMP% directory I (even Windows developer) have never seen and find it very unusual and inflexible.

I would expect from a portable program that it does not write to %APPDATA%. Either you can configure everything or the program writes everything inside the directory where it is "installed".

Since I extracted the program from the RAR-sfx and run it from C:\PortableApps\VidCoder\ directory, it probably won't run in portable mode for me. Therefore I don't have to wonder that under %APPDATA%\Local\ there is still a folder "IsolatesStotage". But this data is probably created by the GUI framework used in VidCoder.

As I said, if I see "portable" somewhere, I would assume that....
- the program does not store files or leave entries anywhere in the system (directories or registry), but stores this data either in its own "installation directory" or in configurable directories.

- The program is not portable only if it is in a specific location (%TEMP%). There are admittedly also exceptions here. Some developers regulate this in such a way that their program is always portable, if it was started from an external drive or USB stick. But as I said: If a developer explicitly offers a separate version for download as "portable", I would assume that this special version IS portable - no matter from where it was executed.
Okay, if you say "yes, but you unpacked the RAR exe, that's not in my sense", then you are right, but unfortunately your solution has the consequence that the start of the program always takes a few seconds, because the RAR archive is unpacked again and again into the %TEMP% directory at every execution. This is not very efficient.

Please take this as a hint how one of your users thinks about it. I'm fine with it, because I can log every program installation and remove all paths and reg entries when uninstalling later.

Thanks for VidCoder and the work you put into it. I appreciate the program very much.

**Comment by RandomEngy** [comment]
Yeah, manually extracting out files from the portable installer is not supported and I am not surprised that it has weird behavior.

If you are particular about where you want the files to be installed and want quick launches, you can use the zip archive.

**Comment by BinaryRider** [comment]
Oh, ok, but where I find the zip archive?
The only zip archive I found at vidcoder.net is the source of VidCoder. Thanks.
